app/src/main/java/edu/wisc/ece/pinpoint/data/Pin.java

```diff
@@ -0,0 +1,78 @@
+package edu.wisc.ece.pinpoint.data;
+
+import android.util.Log;
+
+import androidx.annotation.NonNull;
+
+import com.google.android.gms.tasks.OnFailureListener;
+import com.google.android.gms.tasks.OnSuccessListener;
+import com.google.firebase.firestore.DocumentReference;
+import com.google.firebase.firestore.FirebaseFirestore;
+
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+
+public class Pin {
+    public enum PinType {
+        TEXT,
+        IMAGE
+    }
+    private static final String TAG = Pin.class.getName();
+    private String authorUID;
+    private String caption;
+    private Date timestamp;
+    // contents will be a text if PinType == TEXT
+    // or a URL to an image if PinType == IMAGE
+    private String contents;
+    private PinType type;
+
+    public Pin(String caption, String authorUID, PinType type, String contents) {
+        this.caption = caption;
+        this.authorUID = authorUID;
+        this.type = type;
+        this.contents = contents;
+        this.timestamp = new Date();
+    }
```

Comment:
I believe that's only the case when using the automatic firestore object translation, not if manually serializing, so it depends what pattern we want to go with.

Reply:
I feel like we should use Firestore's automatic translation since it looks a lot cleaner and there are less areas for mistakes.

Reply:
I'm leaning towards building our own maps / documents instead of using the firestore ORM since it offers more flexibility, i.e. we can organize our Java objects however we want and then run a serialization function vs having to organize e.g. nested objects in the way firestore wants. It also allows us to add fields to the object that we might not want to store in firestore.

Reply:
Ah yeah that's true. I definitely agree that properties we don't push to firestore will be useful moving forward. Let's go with that then.